### PR TITLE
Rename the resource group parameter 'memory_limit' to 'memory_quota'

### PR DESCRIPTION
**Pull request by SmartKeyerror**
`CREATE RESOURCE GROUP` and `ALTER RESOURCE GROUP`, both 6X and 7X have the same 
parameter name, though the meaning for them is totally different (7X is an absolute number and 
6X is a percentage of memory).

So we rename 7X parameter to `memory_quota` to clearly convey its difference from 6X 
and also to convey the units as well for ease of use. 

```
CREATE RESOURCE GROUP rg_memory_test WITH(memory_quota=-1, cpu_max_percent=20, concurrency=2);
ALTER RESOURCE GROUP rg_memory_test SET memory_quota 1000;
```

### CODE REVIEW

**Comment by adam8157** [comment]
Please sync with the reporter with the name to change to, and have a separate commit addressing it in docs.

**Comment by ashwinstar** [comment]
Sorry for iteration on the name of the field (its hard naming we all know)... thinking more seems we can just have `memory_quota` instead of `memory_limit_quota` simple to read/write/understand. Feel free to seek opinions from others as well and go ahead with whichever you feels best. I am fine with both.

File: src/backend/parser/gram.y [hunk 1]
```diff
@@ -1651,9 +1651,9 @@ OptResourceGroupElem:
 				{
 					$$ = makeDefElem("cpuset", (Node *) makeString($2), @1);
 				}
-			| MEMORY_LIMIT SignedIconst
+			| MEMORY_QUOTA SignedIconst
 				{
-					$$ = makeDefElem("memory_limit", (Node *) makeInteger($2), @1);
+					$$ = makeDefElem("MEMORY_QUOTA", (Node *) makeInteger($2), @1);
```
Comment (adam8157) [inline src/backend/parser/gram.y:1656]: I believe it's case-sensitive.
```
else if (strcmp(defname, "memory_quota") == 0)
```

**Review by adam8157 — COMMENTED** [review]
Better to have a pipeline to test, even it's only a changing name PR.

**Review by adam8157 — APPROVED** [review]

**Pull request merged**